Marlin/src/lcd/dwin/e3v2/dwin.cpp

```diff
@@ -2321,9 +2321,9 @@ void HMI_Prepare() {
         DWIN_Draw_FloatValue(true, true, 0, font8x16, Color_White, Color_Bg_Black, 3, 1, 216, MBASE(2), current_position.y * MINUNITMULT);
         DWIN_Draw_FloatValue(true, true, 0, font8x16, Color_White, Color_Bg_Black, 3, 1, 216, MBASE(3), current_position.z * MINUNITMULT);
         #if HAS_HOTEND
-          queue.inject_P(PSTR("G92 E0"));
```

Comment:
It bothers me that I cannot explain why this wasn't working. If nobody can reply with a good explanation I will try to debug a little more.

Reply:
I figured out why this didn't work. `G92` only adjust the current position if the needed asjustment is `!NEAR_ZERO`.

This code sets current_position before the injected command runs, which will prevent the planner from ever being updated.

My fix seems appropriate, since this jogging code already manipulates `current_position`. Within one mechanism it probably doesn't make sense to use g-code calls for some position-related tasks and internal calls for others.